dev_docs/deep_zoom.md

```diff
@@ -0,0 +1,32 @@
+# Spatial Mode Deep Zoom Feature
+
+The Deep Zoom feature in CELLxGENE Explorer enhances the spatial exploration of datasets by allowing users to zoom in and out of high-resolution images smoothly. This functionality is critical for detailed examination of spatial data, such as those from Slide-seq V2 or Visium, providing fine-grained insight into tissue samples and cellular interactions.
+
+It is built on the OpenSeadragon (OSD) JavaScript library, integrated into the Explorer’s frontend to deliver a responsive zooming experience. The backend handles the generation and storage of deep zoom image tiles, making it efficient to load and view high-resolution spatial data across environments (dev, staging, prod).
+
+## Frontend Integration (Explorer FE)
+
+- [OpenSeadragon (OSD)](https://openseadragon.github.io/): The OSD library is imported into the frontend to handle interactive zoom and pan operations. OSD dynamically loads image tiles based on the user’s zoom level and position.
+- Zoom & Pan Synchronization: Zooming and panning are kept in sync with spatial layers (e.g., dot layers) in the Explorer to ensure smooth interaction across different dataset views.
+- Dynamic Resource Loading: URLs for the image tiles are generated based on the current environment (dev, staging, prod), ensuring the correct assets are fetched dynamically.
+
+## Backend Image Processing & Storage
+
```

Comment:
There's also an API endpoint that sends some metadata like image_width, image_height, resolution, scaleref etc. used in dot size calculation. 

Another endpoint normalizes spatial embedding so it aligns with the image underlay.